docs/pubsubsource/README.md

```diff
@@ -0,0 +1,199 @@
+most useful as a bridge from other Google Cloud services, such as
+[Cloud Storage](https://cloud.google.com/storage/docs/pubsub-notifications),
+[IoT Core](https://cloud.google.com/iot/docs/how-tos/devices) and
+[Cloud Scheduler](https://cloud.google.com/scheduler/docs/creating#).
+
+## Prerequisites
+
+1. Create a
+   [Google Cloud project](https://cloud.google.com/resource-manager/docs/creating-managing-projects)
+   and install the `gcloud` CLI and run `gcloud auth login`. This sample will
+   use a mix of `gcloud` and `kubectl` commands. The rest of the sample assumes
+   that you've set the `$PROJECT_ID` environment variable to your Google Cloud
+   project id, and also set your project ID as default using
+   `gcloud config set project $PROJECT_ID`.
+
+1. [Install Cloud Run Events](../install).
+
+1. Enable the `Cloud Pub/Sub API` on your project:
+
+   ```shell
+   gcloud services enable pubsub.googleapis.com
+   ```
+
+1. Create a
+   [Google Cloud Service Account](https://console.cloud.google.com/iam-admin/serviceaccounts/project).
+   This sample creates one service account for both registration and receiving
+   messages, but you can also create a separate service account for receiving
+   messages if you want additional privilege separation.
+
+   1. Create a new service account named `cloudrunevents-pubsubsource` with the
+      following command:
+      ```shell
+      gcloud iam service-accounts create cloudrunevents-pubsubsource
+      ```
+   1. Give that Service Account the `Pub/Sub Editor` role on your Google Cloud project:
+      ```shell
+      gcloud projects add-iam-policy-binding $PROJECT_ID \
+        --member=serviceAccount:cloudrunevents-pubsubsource@$PROJECT_ID.iam.gserviceaccount.com \
+        --role roles/pubsub.editor
+      ```
+   1. Download a new JSON private key for that Service Account. **Be sure not to
+      check this key into source control!**
+      ```shell
+      gcloud iam service-accounts keys create cloudrunevents-pubsubsource.json \
+        --iam-account=cloudrunevents-pubsubsource@$PROJECT_ID.iam.gserviceaccount.com
+      ```
+   1. Create a secret on the kubernetes cluster with the downloaded key:
+
+      ```shell
+          # The secret should not already exist, so just try to create it.
+          kubectl --namespace default create secret generic google-cloud-key --from-file=key.json=cloudrunevents-pubsubsource.json
+      ```
+      
+      `google-cloud-key` and `key.json` are pre-configured values in
+      [`pubsub-source.yaml`](./pubsub-source.yaml).
+
+## Deployment
+
+1. Create a Cloud Pub/Sub Topic.
+
+   ```shell
+   export TOPIC_NAME=testing
+   gcloud pubsub topics create $TOPIC_NAME
+   ```
+
+1. Update `TOPIC_NAME` in the [`pubsub-source.yaml`](./pubsub-source.yaml) and
+   apply it.
+
+   If you're in the samples directory, you can replace `TOPIC_NAME` and apply in
+   one command:
+
+   ```shell
+   sed "s/\TOPIC_NAME/$TOPIC_NAME/g" pubsub-source.yaml | \
```

Comment:
bit of a nit here. This assumes you have actually downloaded the files (or cloned the git repo). It would be nice if we could have a 'curl' way to get these files so that we don't raise the bar unnecessarily for just kicking the tires.

Reply:
agreed, as a follow up?

Reply:
https://github.com/GoogleCloudPlatform/cloud-run-events/issues/16